### PR TITLE
Fix AtlasPacker ignoring semi-transparent pixels

### LINKED ISSUE #33106: Importing TextureAtlas cuts out the sprites
**Godot version:**
3.2 alpha3

**OS/device including version:**
Ubuntu 18.04.3 LTS

**Issue description:**
When importing the sprites as TextureAtlas it cuts out the sprites like in the picture.
Every sprite is affected, some more than others. Pic has extreme examples.
Untitled

**Steps to reproduce:**
Import some sprites

### PR DESCRIPTION
**Pull request by ericrybick**
fixes #33106

### CODE REVIEW

**Comment by groud** [comment]
The variable should be renamed, that's not what "opaque" mean anymore.
Also, changing something there might break compatibility and break other things into the code (the Image API is used in a lot of places). This should be verified.

Maybe the opacity threshold should be provided as a parameter of the function instead ?

**Comment by ericrybick** [comment]
Thanks for reviewing!
I've removed the variable and provided the threshold as parameter as you suggested.

The function has just two callers. The AtlasPacker and `EditorAssetLibraryItemDescription::set_image`.

The latter creates the play logo overlay for videos in the Asset Library. While in my tests it works fine with alpha set to `0`, I've set it back to `0.99` just to be sure that nothing breaks.

File: editor/plugins/asset_library_editor_plugin.cpp [hunk 1]
```diff
@@ -167,7 +167,7 @@ void EditorAssetLibraryItemDescription::set_image(int p_type, int p_index, const
 						// Overlay and thumbnail need the same format for `blend_rect` to work.
 						thumbnail->convert(Image::FORMAT_RGBA8);
 						thumbnail->lock();
-						thumbnail->blend_rect(overlay, overlay->get_used_rect(), overlay_pos);
+						thumbnail->blend_rect(overlay, overlay->get_used_rect(0), overlay_pos);
```
Comment (akien-mga) [inline editor/plugins/asset_library_editor_plugin.cpp]: Shouldn't this be 0.99 to preserve the previous behavior?

File: core/image.cpp [hunk 2]
```diff
@@ -2027,8 +2027,7 @@ Rect2 Image::get_used_rect() const {
 	for (int j = 0; j < height; j++) {
 		for (int i = 0; i < width; i++) {
 
-			bool opaque = get_pixel(i, j).a > 0.99;
-			if (!opaque)
+			if (!(get_pixel(i, j).a > 0))
```
Comment (akien-mga) [inline core/image.cpp]: Could also be `if (get_pixel(i, j).a == 0)` directly.

**Review by akien-mga — APPROVED** [review]
As discussed on IRC, this actually fixes a buggy implementation of `get_used_rect()` and the only two downstream users in the editor now work properly.

This is exposed to scripting languages, so while it doesn't break user code, it might change the logic for their use cases. Yet the existing description of the function supports this implementation, so I just asked @ericrybick to clarify the implementation in the docs.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**